conf/nomad/presto.hcl

```diff
@@ -38,30 +44,70 @@ job "${nomad_job_name}" {
           }
         }
       }
-      check {
-        task     = "server"
-        name     = "presto-hive-availability"
-        type     = "script"
-        command  = "presto"
-        args     = ["--execute", "SHOW TABLES IN hive.default"]
-        interval = "30s"
-        timeout  = "15s"
+    }
+    service {
+      name = "%{ if node_type == "coordinator" }${service_name}%{ else }${service_name}-worker%{ endif }"
+      tags = [
+      "$${NOMAD_PORT_connect}",
+      "%{ if node_type == "coordinator" }coordinator%{ else }worker%{ endif }"
+      ]
+      port = "connect"
+      task = "server"
+      connect {
+        native = true
       }
       check {
-        expose   = true
         name     = "presto-info"
         type     = "http"
+        protocol = "https"
+        tls_skip_verify = true
         path     = "/v1/info"
         interval = "10s"
         timeout  = "2s"
       }
       check {
-        expose   = true
-        name     = "presto-node"
-        type     = "http"
-        path     = "/v1/node"
-        interval = "10s"
-        timeout  = "2s"
+        task     = "server"
+        name     = "presto-started"
+        type     = "script"
+        command  = "/bin/sh"
+        failures_before_critical = 2
+        args     = ["-c", "curl -s -k https://127.0.0.1:$${NOMAD_PORT_connect}/v1/info | grep -Po '(?<=)\"starting\":false(?=,)[^,]*'"]
+        interval = "5s"
+        timeout  = "30s"
+      }
+    }
+
+    task "certificate-handler" {
+      lifecycle {
+        hook    = "prestart"
+        sidecar = true
+      }
+      driver = "docker"
+      config {
+        image = "fredrikhgrelland/alpine-jdk11-openssl:0.1.0"
+        entrypoint = ["/bin/sh"]
+                # This task is built to get certificate, ca and key from consul via. nomad template stanza and build a java keystore.
+                # The container will run a shell and execute the following commands before waiting forever ´tail -f /dev/null´
+                # 1. Create a pkcs12 ´local/presto.p12´ in openssl based on leaf key and certificate provided by the template stanza.
+                # 2. Import ´local/presto.p12´ using keytool and output ´local/presto.jks´
+                # 3. Import ´local/presto.jks´ using keytool, add root as CA and output ´alloc/presto.jks´
+                # 4. Keep running the container `tail -f /dev/null`
+                # Now we can use this ´/alloc/presto.jks´ from any task in the group(alloc)
```

Comment:
The formatting is out of whack on my screen
![image](https://user-images.githubusercontent.com/51820995/93459738-7821cd80-f8e2-11ea-8aef-b357342a95a5.png)